# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1534173394 14400
#      Mon Aug 13 11:16:34 2018 -0400
# Node ID 4e658452f6c7b680b2142ac406b0381ec46b9916
# Parent  cf2914e29d930f346756f76db6b6c1db16720199
add -std=legacy to FFLAGS if GNU compiler accepts it (bug #54390)

* configure.ac: If using GNU Fortran compiler, add -std=legacy to
FFLAGS if it is a valid option.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -836,16 +836,20 @@ AC_DEFINE_UNQUOTED(OCTAVE_SIZEOF_F77_INT
   [Define to the size of the octave_f77_int_type (8 or 4).])
 
 OCTAVE_F77_FLAG([-ffloat-store], [
   AC_MSG_RESULT([setting F77_FLOAT_STORE_FLAG to -ffloat-store])
   F77_FLOAT_STORE_FLAG=-ffloat-store
   AC_SUBST(F77_FLOAT_STORE_FLAG)
 ])
 
+if test $ac_cv_f77_compiler_gnu = yes; then
+  OCTAVE_F77_FLAG([-std=legacy])
+fi
+
 BUILD_EXTERNAL_LIBXERBLA=
 case $host_os in
   msdosmsvc | mingw*)
     BUILD_EXTERNAL_LIBXERBLA=ues
   ;;
 esac
 
 AM_CONDITIONAL([AMCOND_BUILD_EXTERNAL_LIBXERBLA],
